File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int prefix = 1;
        vector<int> prefixV(nums.size(), 1);
        for(int i = 0; i < nums.size(); ++i){
            prefixV[i] = prefix;
            prefix *= nums[i];
        }

        int pofix = 1;
        vector<int> res(nums.size(), 1);
        for(int i = nums.size() - 1; i >= 0; --i){
            res[i] = pofix * prefixV[i];
            pofix *= nums[i];
        }
        return res;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        mp = defaultdict(int)
        for num in nums:
            mp[num] += 1
        
        buckets = defaultdict(list)
        for key, val in mp.items():
            buckets[val].append(key)
        res = []
        for i in range(len(nums), 0, -1):
            for j in range(len(buckets[i])):
                if len(res) == k:
                    return res
                res.append(buckets[i][j])
        
        return res